src/ergtable.h

```diff
@@ -0,0 +1,140 @@
+#ifndef ERGTABLE_H
+#define ERGTABLE_H
+
+#include <QList>
+#include <QSettings>
+#include <QObject>
+#include <cmath>
+
+#include <QMetaType>
+#include <QDebug>
+
+class DataPoint {
+public:
+    double cadence = 0; // RPM, default initialized
+    double wattage = 0; // Watts, default initialized
+    double resistance = 0; // Some unit, default initialized
+
+    DataPoint() = default; // Default constructor
+
+    DataPoint(double c, double w, double r) : cadence(c), wattage(w), resistance(r) {} // Custom constructor
+};
+
+Q_DECLARE_METATYPE(DataPoint)
+
+class ergTable : public QObject {
+    Q_OBJECT
+
+public:
+    ergTable(QObject *parent = nullptr) : QObject(parent) {
+        loadSettings();
+    }
+
+    ~ergTable() {
+        saveSettings();
+    }
+
+    void collectData(double cadence, double wattage, double resistance) {
+        if (wattage > 0 && !dataPointExists(cadence, wattage, resistance)) {
+            qDebug() << "newPointAdded" << "C" << cadence << "W" << wattage << "R" << resistance;
+            DataPoint point(cadence, wattage, resistance);
+            dataTable.append(point);
+            saveDataPoint(point); // Save each new point to QSettings
+        } else {
+            qDebug() << "discarded" << "C" << cadence << "W" << wattage << "R" << resistance;
+        }
+    }
+
+    double estimateWattage(double givenCadence, double givenResistance) {
+        // Placeholder for closest points. Initialize with max possible difference.
+        double lowerDiff = std::numeric_limits<double>::max();
+        double upperDiff = std::numeric_limits<double>::max();
+        DataPoint lowerPoint, upperPoint;
+
+        for (const DataPoint& point : dataTable) {
```

Comment:
Can you be sure this dataTable will be consulted and updated always on the same thread?

Reply:
Actually nope, do you think about a lock mechanism?

Reply:
I'm using Qt's various mutexes and QMutexLocker in #899 .

Reply:
> I'm using Qt's various mutexes and QMutexLocker in #899 .

Have a classmember of type QMutex or QRecursiveMutex if you might need multiple locks. 
Use a QMutexLocker to lock it AND make sure it gets unlocked if there's an exception.